docs/enhancements/agent-upgrade.md

```diff
@@ -0,0 +1,189 @@
+---
+title: agent-upgrade
+authors:
+  - "@jhernand"
+creation-date: 2022-06-13
+last-updated: 2022-06-13
+---
+
+# Agent upgrade
+
+## Summary
+
+Usually the time since the agent is started and the cluster is installed is
+short. But in some situations it may be long, even many days, and during that
+long time the service may be upgraded and require a newer version of the agent.
+The current mechanism to perform that agent upgrade is to reboot the host. That
```

Comment:
There is no current mechanism. We just work with older agent versions and pray nothing goes wrong

Reply:
Done.